java/arcs/android/devtools/DevWebSocket.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.devtools
+
+import android.content.ContentValues.TAG
+import android.util.Log
+import fi.iki.elonen.NanoHTTPD
+import fi.iki.elonen.NanoWSD
+import java.io.IOException
+
+/**
+ * An extension of [NanoWSD] for devtools to connect Arcs to a remote device for debugging.
+ */
+class DevWebSocket : NanoWSD(12345) {
+    private lateinit var wsdSocket: WsdSocket
+
+    /**
+     * Send a string to the client.
+     */
+    fun send(msg: String) {
+        wsdSocket.send(msg)
+    }
+
+    override fun openWebSocket(ihttpSession: NanoHTTPD.IHTTPSession?): WebSocket {
```

Comment:
`override fun openWebSocket(ihttpSession: NanoHTTPD.IHTTPSession?) = WsdSocket(ihttpSession)` ?

Reply:
This doesn't work because we need to initialize wsdSocket for the send method.